services/venice-router/src/main/java/com/linkedin/venice/router/throttle/ReadRequestThrottler.java

```diff
@@ -88,16 +89,16 @@ public ReadRequestThrottler(
     this.storeQuotaCheckTimeWindow = storeQuotaCheckTimeWindow;
     this.stats = stats;
     this.maxRouterReadCapacity = maxRouterReadCapacity;
-    this.zkRoutersManager.subscribeRouterCountChangedEvent(this);
     this.lastRouterCount = zkRoutersManager.getExpectedRoutersCount();
     this.perStoreRouterQuotaBuffer = perStoreRouterQuotaBuffer;
     this.idealTotalQuotaPerRouter = calculateIdealTotalQuotaPerRouter();
     this.isNoopThrottlerEnabled = false;
 
     /** Calling {@link #buildAllStoreReadThrottlers()} should be done after all internal state is initialized */
-    this.storesThrottlers = new AtomicReference<>(buildAllStoreReadThrottlers());
+    this.storesThrottlers.set(buildAllStoreReadThrottlers());
```

Comment:
Even if there were multiple threads constructing instances of `ReadRequestThrottler`, there would be only one of these threads executing the constructor code for any given instance, so that is not the cause of the race condition. The cause is the listener registration which you pointed out in your other comment.